distributed/worker.py

```diff
@@ -268,7 +273,9 @@ def heartbeat(self):
                     in_memory=len(self.data),
                     ready=len(self.ready),
                     in_flight=len(self.in_flight_tasks),
-                    **self.monitor.recent())
+                    custom_metrics_names=list(custom_metrics.keys()),
```

Comment:
I ended up adding the custom_metric_names as another field in the call-back. With this you know the metrics which were added, which allows to make the code downstream a lot simpler. Maybe there is a better way of passing the metrics names around though?

Reply:
I don't understand, why isn't the dictionary passed in `custom_metrics` above sufficient?

Reply:
`custom_metrics_names` is useful for the downstream code to figure out which one are the "additional custom metrics" that should go in the secondary table. Without it, you need another way to figure that out because all of the metrics have the same status in the `worker_state.info` dictionary. One is to have a long hard-coded list metric to ignore (i.e. all the ones that comes from `SystemMonitor` plus some others IIRC).